### DISCUSSION TITLE
Is there a way to delete or hide old/renamed Workflows? (Reviving #26256)

### DISCUSSION THREAD

**Discussion by egk-ts** [community/community#40350] [Actions] [answered]
### Select Topic Area

Product Feedback

### Body

As already asked in #26256; 

> I made a Workflow using the Node CI template and then later renamed it.  Now my Workflow list has “Node CI” in it, but that doesnt even exist anymore.  Is there some way for me to hide it?  Or is it there forever?

#26256 was marked as answered three years ago with the following response;

> You can’t remove it yet, but it won’t be there forever.  You can’t delete it yet but we’ll have functionality to delete this in the future.

Based on the continous stream of "+1" on #26256 and the comments by many, it appears that this issue might still be important to many and it might not received due attention, perhaps due to it being marked as "Answered". 

Is it possible to give this some priority or provide a more tangible estimate than "eventually"?

**Comment by reneyo** [comment]
\+ 1

**Reply by EntropyAu** [reply]
+1

**Reply by jd-yolabs** [reply]
+1

**Reply by mkenigs** [reply]
+1

**Reply by Brian-Azizi** [reply]
+1

**Reply by pcornelissen** [reply]
Please use the emoticons on the existing comments for reactions and stop the +1 mania

**Reply by AndreyBig567** [reply]
+1

**Reply by dgandiaga** [reply]
+1

**Reply by jinluchang** [reply]
+1

**Reply by povisenko** [reply]
+1

**Reply by edtep-relayfi** [reply]
+1

**Comment by ndrsllwngr** [comment]
- Having the option to remove ghost workflows with (1) zero runs anymore and (2) no .yaml configuration file would make it easier to navigate a rather big project which evolved over time.
- Furthermore, this would support us in exploring new workflows without having to bear potential ghost workflows in mind, which might spam the GitHub Actions page.

<img width="1624" alt="Screenshot 2022-10-14 at 00 18 12" src="https://user-images.githubusercontent.com/10077258/195721259-520b2293-4248-4107-8bce-684903b060f5.png">

> repost from https://github.com/orgs/community/discussions/26256#discussioncomment-3874137

**Comment by this-sam** [comment]
I chimed in on the original thread but adding my support here as well. Deleting or hiding unused/obsolete workflows would be a huge help for my org.

**Comment by supmanyu** [comment]
This is a critical feature missing from github actions, if a new dev starts work on a repo, how will they know which job to run!?

![image](https://user-images.githubusercontent.com/13700827/205493918-b2ddd81b-0930-42a6-8c46-c89bb6a4a561.png)

**Comment by santhosh-vikraman** [comment]
This feature is a much needed one

**Comment by TonyWhiteSMS** [comment]
Yep the ability to delete or archive is needed...

**Comment by bokicks283** [comment]
This is so silly plus it makes there documentation incorrect... which is kind of infuriating. https://docs.github.com/en/actions/managing-workflow-runs/deleting-a-workflow-run

**Comment by justinjbird** [comment]
festive +1

**Comment by tyler1314371** [comment, marked as answer]
Apparently if you delete all the runs for a given workflow, the workflow will then disappear from the list as well. You can delete it manually from the UI or this is the script that I used to clear all runs for a given workflow. Hope it helps!

Note that it needs`gh` and `jq` cli installed
```
OWNER=<your user/org name>
REPO=<repo name>

# list workflows
gh api -X GET /repos/$OWNER/$REPO/actions/workflows | jq '.workflows[] | .name,.id'

# copy the ID of the workflow you want to clear and set it
WORKFLOW_ID=<workflow id>

# list runs
gh api -X GET /repos/$OWNER/$REPO/actions/workflows/$WORKFLOW_ID/runs | jq '.workflow_runs[] | .id'

# delete runs (you'll have to run this multiple times if there's many because of pagination)
gh api -X GET /repos/$OWNER/$REPO/actions/workflows/$WORKFLOW_ID/runs | jq '.workflow_runs[] | .id' | xargs -I{} gh api -X DELETE /repos/$OWNER/$REPO/actions/runs/{}
```

**Reply by egk-ts** [reply]
I can confirm; This works. 

You may augment the last line as follows, to keep running it indefinetely and save yourself from doing it manually (Note that the `while` loop doesn't break when done because `gh` still returns exit code 0, so look wait for "gh: Not Found (HTTP 404)" and break it with `ctrl+c`)

    while gh api -X GET /repos/$OWNER/$REPO/actions/workflows/$WORKFLOW_ID/runs | jq '.workflow_runs[] | .id' | xargs -I{} gh api -X DELETE /repos/$OWNER/$REPO/actions/runs/{}; do echo ...; sleep 1; done

**Reply by Papooch** [reply]
Every API call jumps into a pager for me and I need to press `Q` to close the preview and continue (`gh` version `2.4.0`). This can be fixed by adding the `--silent` option to the DELETE call.

**Comment by piotrekkr** [comment]
Instead of adding this to UI or API they added an [example script in docs](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/using-workflow-run-logs#example-script) :trollface:
